.github/workflows/payments.yaml

```
@@ -0,0 +1,39 @@
+name: Build payments service backend
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - "payments-be/**"
+  pull_request:
+    branches: ["main"]
+    paths:
+      - "payments-be/**"
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      working-directory: payments-be
+
+    defaults:
+      run:
+        working-directory: ${{ env.working-directory }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      - name: Set up gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Build with Gradle
+        run: ./gradlew build
```

```suggestion
    steps:
    - uses: actions/checkout@v3
    - name: Set up JDK 17
      uses: actions/setup-java@v3
      with:
        java-version: '17'
        distribution: 'temurin'
        cache: maven
    - name: Build with Maven
      run: mvn package
```

This will run build as werll as tests. Cashe is so that maven dependency plugin can make the thing run a little between faster between possible reruns.